File: 1512-number-of-good-pairs/1512-number-of-good-pairs.go
func numIdenticalPairs(nums []int) int {
    hashMap := make(map[int]int)
    answer := 0
    
    for i := range nums {
        _, ok := hashMap[nums[i]]
        if !ok {
            hashMap[nums[i]] = 0
        }
        hashMap[nums[i]]++
        if hashMap[nums[i]] > 1 {
            answer += hashMap[nums[i]] - 1
        }
    }
    
    return answer
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.ts
function numIdenticalPairs(nums: number[]): number {
    const hashMap = new Map<number, number>()
    var ans: number = 0
    
    nums.forEach((value:number) => {
        if (hashMap.has(value)) {
            var count: number = hashMap.get(value)
            hashMap.set(value,++count)
            if (count>1) {
                ans += count-1
            }
        } else {
            hashMap.set(value, 1)
        }
    })
    return ans
};